mongo/src/main/java/com/redhat/lightblue/mongo/crud/Translator.java

```diff
@@ -1258,11 +1260,12 @@ private BasicDBObject objectToBson(JsonNodeCursor cursor, EntityMetadata md) {
             Path path = cursor.getCurrentPath();
             JsonNode node = cursor.getCurrentNode();
             LOGGER.debug("field: {}", path);
-            FieldTreeNode fieldMdNode = md.resolve(path);
-            if (fieldMdNode == null) {
-                throw Error.get(ERR_INVALID_FIELD, path.toString());
+            FieldTreeNode fieldMdNode = null;
+            try {
+                fieldMdNode = md.resolve(path);
+            } catch (Exception e) {
+                return null;
```

Comment:
Null won't be returned. The old code checks null and throws an exception. Null check is unnecessary, resolve already throws the exception. so getting rid of the null check in the old code should be sufficient.